### PR TITLE
Object.is documentation

### PR DESCRIPTION
**Pull request by jmagaram**
Documentation for Object.is. Also some tests mostly so I could verify how it all works. Took the liberty of documenting some of `==` and `===` at the same time. Read through the Mozilla docs and tried to summarize.

### CODE REVIEW

**Comment by jmagaram** [comment]
According to Mozilla docs "In general, the only time Object.is special behavior towards zeros is likely to be of interest is in the pursuit of certain meta-programming schemes...suggested to avoid".

**Comment by zth** [comment]
This is looking great! Out traveling, but will try to review properly as soon as I can.

File: src/Core__Object.res [hunk 1]
```diff
@@ -1,6 +1,37 @@
...
 
-@val external is: ('a, 'b) => bool = "Object.is"
+/**
+`is` determines if two objects are identical in all contexts. Objects, arrays, records, and other non-primitives are only identical if they reference the **exact** same object in memory. Primitives like ints, floats, and strings are identical if they have the same value. `+0` and `-0` are distinct. NaN is equal to itself. See [Object.is on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is)
+
+In most scenarios use `==` or `===` or the custom `equals` function (if provided) for the type.
+
+The `==` operator [is different in ReScript than Javascript](https://rescript-lang.org/docs/manual/latest/overview#boolean). Arrays, records and other non-primitives are equal if they have the same contents (deep equality).
+
+In ReScript, the `===` operator performs a strict equality check, like Javascript, and is similar but not identical to `is`
```
Comment (glennsl) [inline src/Core__Object.res:10]: Describing other functions and operators seem outside the scope of reference documentation. It will quickly become hard to maintain the documentation if it gets scattered across other definitions.

File: src/Core__Object.res [hunk 2]
```diff
@@ -1,6 +1,37 @@
...
+
+let fruit = {"name": "Apple" }
+Object.is(fruit, fruit) // true
+Object.is(fruit, {"name": "Apple" }) // false
+fruit == {"name": "Apple" } // true
+fruit === {"name": "Apple" } // false
+```
+*/
+@val
+external is: ('a, 'b) => bool = "Object.is"
```
Comment (glennsl) [inline src/Core__Object.res:34]: I'm not sure this type signature makes sense. By definition you'd think that objects of different types wouldn't be considered identical, and you might as well weed that out at compile time.

There are of course exceptions, such as JSON-encoded primitives, but better to require explicitness than cater to convenience for weird edge cases I think.

I would therefore suggest this instead:

```suggestion
external is: ('a, 'a) => bool = "Object.is"
```

File: src/Core__Object.res (hunk 1 above)
Comment (jmagaram) [reply src/Core__Object.res:10]: Ok thanks for the feedback.

**Comment by jmagaram** [comment]
Git problems here. I broke the tests because I uploaded the type signature change but forgot to remove tests that compare different types. I don't know how to fix it. Sorry! Do I do a git pull rebase? VS Code wants me to do a Sync but it isn't working. Not sure what steps to do in VS Code. Don't want to lose work here. Or you can fix it if it is easy for you.

**Comment by jmagaram** [comment]
Ok fixed it.

**Comment by jmagaram** [comment]
Consolidated into https://github.com/rescript-association/rescript-core/pull/117
